result_test.go

```diff
@@ -409,11 +409,13 @@ func TestToMapBy(t *testing.T) {
 
 func TestToSlice(t *testing.T) {
```

Comment:
Hmm the godoc for `ToSlice` doesn't really tell me that it starts overwriting the given slice from index=0. Is that the behavior? Perhaps we should rewrite `ToSlice` godoc?